single-host-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/singlehostclient/SingleIpHttpRequest.java

```diff
@@ -47,7 +56,12 @@ public boolean expectContinue() {
 
     @Override
     public URI uri() {
-        return httpRequest.uri();
+        final URI uri = httpRequest.uri();
+        try {
+            return new URI(uri.getScheme(), uri.getUserInfo(), hostAddress.getHostAddress(), uri.getPort(), uri.getPath(), uri.getQuery(), uri.getFragment());
```

Comment:
This is forcing the URL to be the host address instead of requiring the client to set it manually.